extras/create_snapshots_reports_scorecard.py

```diff
@@ -79,6 +79,27 @@
 report_durations = list()
 rd_lock = threading.Lock()
 
+# Global variables and thread locks for third-party snapshots and reports
+# Note that snapshots_to_generate and reports_to_generate (and their associated
+# locks) are reused for third-party snapshots and reports, so we don't need
+# variables and locks for them below.
+successful_tp_snapshots = list()
+stps_lock = threading.Lock()
```

Comment:
Since third-party work is done after first-party work we could probably reuse a number of these locks and just do

```python
with lock_in_question:
    if third_party:
        # Do third-party thing
    else:
        # Do first-party thing
```

Just a thought since adding a single boolean value check to the critical section is negligible in my mind.

Reply:
I see your point.  I created those third-party locks mainly to reduce confusion and try to keep things named consistently, but since I'm already re-using some of the things (`snapshots_to_generate` and `reports_to_generate`), what's a few more?  I will ponder.

Reply:
I have come to the conclusion that I agree with your suggestion, which I put in https://github.com/cisagov/cyhy-reports/pull/118/commits/dea1135e1c791a79494dd906799d262b439fa564.  Excellent thinking!